src/http/client.cr

```diff
@@ -807,48 +799,39 @@ class HTTP::Client
     end
   end
 
-  {% if flag?(:without_openssl) %}
-    protected def self.tls_flag(uri, context : Nil)
-      scheme = uri.scheme
+  protected def self.tls_flag(uri, context : TLS) : TLS
+    scheme = uri.scheme
+    raise ArgumentError.new("Missing scheme: #{uri}") if !scheme
+    {% begin %}
```

Comment:
The complex `case` construction seems to complicated to me. I'd prefer two completely separate branches with duplicate `else` behaviour. It's a simple `raise` expression. A little duplication is fine.